### DISCUSSION TITLE
请问如何重新指定文件关联的工具链

### DISCUSSION THREAD

**Discussion by YdrMaster** [xmake-io/xmake#6000] [Q&A] [answered]
我的需求是我希望用我指定的非 NVCC 的工具链编译添加的 *.cu 源码。我设置了一个 standalone 工具链，但 target 中添加的 *.cu 文件还是会自动去找 Cuda SDK 而不是我设置的自定义工具链。怎么修改这个行为？

**Comment by waruqi** [comment]
set_toolchains or `xmake f --toolchains=xxx`

**Reply by YdrMaster** [reply]
这个没有用啊，我设置的编译器是一个 clang/clang++ 的，它似乎关联不上 *.cu 文件。如果改名叫 *.cpp 就正常，保持 *.cu 的话就会继续报没有 Cuda SDK 的错误

**Reply by waruqi** [reply]
目前只能 nvcc 指定 ccbin 去切到 其他编译器 

```console
xmake f --cu-ccbin=clang
```

**Reply by YdrMaster** [reply]
ccbin 指定的是 host 编译器吧。我是说我要用一个魔改的 clang 编译 cu 文件，完全不使用 nvcc 要怎么做。现在这个行为会被 xmake 阻止，xmake 不会吧 cu 文件发送给我通过自定义工具链指定的编译器，只能找 nvcc

**Reply by YdrMaster** [reply, marked as answer]
我知道了，可以用 rule 直接覆盖文件的编译逻辑
